index.rst

```diff
@@ -41,40 +41,46 @@ with version control systems.
 File Format
 ===========
 
-EditorConfig files use an INI format that is compatible with the format used
-by `Python configparser Library`_. In an EditorConfig file (usually named
-``.editorconfig``), all beginning whitespace in each line is ignored. Each
-line must be one of the following:
-
-- Empty Line: An empty string.
-- Comment: A line starting with a ``;`` or a ``#``.
-- Section Title: A line that starts with a ``[``, and that ends with a ``]``.
-- Assignment: A line separated by an `=` into two parts.
-
-Any line that is not one of the above is invalid. "Inline comment" in
-appearance, i.e., a line that has a ``;`` or a ``#`` in the middle, is
-undefined. For convenience, we also define the following terminologies:
-
-- Section Name: The string between the beginning ``[`` and the ending ``]`` in
-  a Section Title.
-- Section: The lines starting from a Section Title until the beginning of the
-  next Section Title or end of file.
-- Property: The part before the first `=` in an Assignment.
-- Value: The part after the first `=` in an Assignment.
-
-In EditorConfig files, the Section Names are filepath globs, similar to the
-format accepted by gitignore. ``[`` and ``]`` are allowed in the Section
-Names. Forward slashes (``/``) are used as path separators and Backslashes
-(``\\``) are not allowed as path separators (even on Windows). Comments should
-always go on individual lines. EditorConfig files should be UTF-8 encoded, with
-either CRLF or LF line separators.
+EditorConfig uses a strict EditorConfig-INI file format. In an EditorConfig
+file (usually named ``.editorconfig``), all beginning whitespace on each line
+is considered irrelevant. Each line must be one of the following:
+
+- Blank: contains only whitespace characters.
+- Comment: starts with a ``;`` or a ``#``.
+   - Inserting a ``#`` or ``;`` after non-whitespace characters in a line
+     (i.e., inline) shall neither be parsed as a comment nor as part of the
+     section name, key or value in which it was inserted. This may change in
+     the future.
```

Comment:
```suggestion
     the future. Users should avoid this uncertain syntax.
```

Reply:
I wouldn't spend too much time on that point, yet. As soon as we get a spec for the ini-file having potentially a roadmap affecting this eventually it will be by far easier to deal with changes to current behavior. My personal target for the first iteration would be to have a fundamental spec to build on.

Reply:
We've already spent a lot of time getting here!

Reply:
True but adding "users should avoid this uncertain syntax" adds implicitly that we already know where we go with that point, yet. And as far as i understood it we do not. And i'd like to avoid starting that discussion yet.

Reply:
But i also have no issues with it - i was fine withe last version as well.

Reply:
> True but adding "users should avoid this uncertain syntax" adds implicitly that we already know where we go with that point, yet. And as far as i understood it we do not. And i'd like to avoid starting that discussion yet.

I thought this would clarify that we don't know what will happen :) Because we do not know what will happen, we kindly ask our users to not engage in doing so.